raster-foundry/raster-foundry
# Update default color space clipping and expose QPs for tuning

## Overview

This PR:
- updates the default color clipping bounds for the auto color correction
- exposes params to set the min/max bounds of color correction on scene rendering

### Checklist

- ~Swagger specification updated~
- ~New tables and queries have appropriate indices added~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~
- ~Any new endpoints have scope validation and are included in the integration test csv~

### Demo

![Screen Shot 2021-04-02 at 6 33 55 PM](https://user-images.githubusercontent.com/16109558/113459297-0e32c680-93e3-11eb-98d4-1ec361c94e82.png)

### Notes
Should we do something special to the decoder of the QPs so that they can only be in range `[0, 99]`? Looks like numbers out of this range can return 500 errors: https://azavea.slack.com/archives/C089LBS6S/p1617742515025900

## Testing Instructions

- Push a `test` branch up
- Grab a spacenet image's S3 location (but can be any image, taking spacenet here as an example bc those images on prod seem dark) from the tile urls [here](https://github.com/raster-foundry/groundwork/blob/develop/src/pages/ProjectCreate/Steps/ImagerySelection/SpacenetImagery/spaceNetProjects.ts), look for a `default: true` project. Or just use an image attached to RF prod project `d95d4348-56da-4358-9af3-e856a329e51c`. Download it to local
- Go to GW staging, enter a campaign, upload the image from the above step. After processing, make sure the new default color clipping makes the image look brighter.
- Grab a z,x,y location and test the QPs using `GET` to `https://tiles.staging.rasterfoundry.com/<RF project ID of this image>/<z>/<x>/<y>?lowerQuantile=0&upperQuantile=99`

Closes https://github.com/azavea/raster-foundry-platform/issues/1215
